File: OrgansDelivery.DAL/Entities/Organ.cs
using OrganStorage.DAL.Interfaces;

namespace OrganStorage.DAL.Entities;

public class Organ : IEntity, IMustHaveTenant, IWithName, IWithDescription
{
    public Guid Id { get; set; }
    public Guid TenantId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime OrganCreationDate { get; set; }
    
    public Guid ContainerId { get; set; }
    public Container Container { get; set; }
    
    public Guid ConditionsId { get; set; }
    public Conditions Conditions { get; set; }

    public ICollection<ConditionsRecord> Records { get; set; }
}

public class OrganDto
{
	public Guid Id { get; set; }
	public string Name { get; set; }
	public string Description { get; set; }
	public DateTime OrganCreationDate { get; set; }
	public Guid ContainerId { get; set; }
	public Guid ConditionsId { get; set; }
}

public class OrganFormValues
{
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime OrganCreationDate { get; set; }
    public Guid ContainerId { get; set; }
	public Guid ConditionsId { get; set; }
}

File: OrgansDelivery.Web/client-app/src/app/models/organ.ts
import { IEntity } from "./common";
import { ConditionsRef } from "./conditions";
import { ContainerRef } from "./container";

export interface Organ extends IEntity {
    name: string;
    description: string;
    organCreationDate: Date;
    container: ContainerRef;
    conditions: ConditionsRef;
}

export interface OrganRef extends IEntity {
    name: string;
    description: string;
    organCreationDate: Date;
    containerId: string;
    conditionsId: string;
}

export interface OrganFormValues {
    name: string;
    description: string;
    organCreationDate: Date;
    containerId: string;
    conditionsId: string;
}
